Drop early-outs for "no boxes" in SVGTextQuery entrypoints

Ahmad-S792 opened the pull request:
#### da2a3ce347f88ab0f3fae1a26a67ce51bc102ba0
<pre>
Drop early-outs for &quot;no boxes&quot; in SVGTextQuery entrypoints

<a href="https://bugs.webkit.org/show_bug.cgi?id=262382">https://bugs.webkit.org/show_bug.cgi?id=262382</a>

Reviewed by Nikolas Zimmermann.

Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=177149

All entrypoints in SVGTextQuery start with check if the text content
element has any boxes. If it doesn&apos;t, a value representing &quot;empty&quot; is
returned. In all cases this value is the same as the default value of the
corresponding query-data. SVGTextQuery::executeQuery itself has no
problems handling this case (and will return &apos;false&apos; as expected.)
Removing these early-outs means a few more cycles is needed in these
&quot;degenerate&quot; cases, although that seems like a small (and hopefully
reasonable) price to pay for less code (and duplication/exit-paths).

* Source/WebCore/rendering/svg/SVGTextQuery.cpp:
(SVGTextQuery::executeQuery):
(SVGTextQuery::numberOfCharacters):
(SVGTextQuery::subStringLength):
(SVGTextQuery::startPositionOfCharacter):
(SVGTextQuery::endPositionOfCharacter):
(SVGTextQuery::rotationOfCharacter):
(SVGTextQuery::extentOfCharacter):
(SVGTextQuery::characterNumberAtPosition):

Canonical link: <a href="https://commits.webkit.org/268702@main">https://commits.webkit.org/268702@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0da5ababa4bd86eda5945e4bc6cb7be21d44dbcc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20335 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20764 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21407 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22233 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18973 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20566 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24022 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20937 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20395 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20554 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20447 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17684 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23083 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17614 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18490 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24778 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18692 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18666 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22710 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16342 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18454 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4908 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22791 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19067 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
r=me, improves readability.

Ahmad-S792 committed:
Drop early-outs for "no boxes" in SVGTextQuery entrypoints

https://bugs.webkit.org/show_bug.cgi?id=262382

Reviewed by Nikolas Zimmermann.

Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=177149

All entrypoints in SVGTextQuery start with check if the text content
element has any boxes. If it doesn't, a value representing "empty" is
returned. In all cases this value is the same as the default value of the
corresponding query-data. SVGTextQuery::executeQuery itself has no
problems handling this case (and will return 'false' as expected.)
Removing these early-outs means a few more cycles is needed in these
"degenerate" cases, although that seems like a small (and hopefully
reasonable) price to pay for less code (and duplication/exit-paths).

* Source/WebCore/rendering/svg/SVGTextQuery.cpp:
(SVGTextQuery::executeQuery):
(SVGTextQuery::numberOfCharacters):
(SVGTextQuery::subStringLength):
(SVGTextQuery::startPositionOfCharacter):
(SVGTextQuery::endPositionOfCharacter):
(SVGTextQuery::rotationOfCharacter):
(SVGTextQuery::extentOfCharacter):
(SVGTextQuery::characterNumberAtPosition):

Canonical link: https://commits.webkit.org/268702@main
